packages/rum-core/src/domain/rumEventsCollection/action/getSelectorFromElement.ts

```diff
@@ -0,0 +1,47 @@
+import { arrayFrom } from '@datadog/browser-core'
+
+export function getSelectorFromElement(targetElement: Element): string {
+  const targetElementSelector = []
+  let element: Element | null = targetElement
+
+  while (element && element.nodeName !== 'HTML') {
+    const idSelector = getIDSelector(element)
+    if (idSelector) {
+      targetElementSelector.unshift(idSelector)
+      break
+    }
+
+    targetElementSelector.unshift(getClassSelector(element) || getPositionSelector(element))
+
+    element = element.parentElement
+  }
+
+  return targetElementSelector.join('>')
+}
+
+function getIDSelector(element: Element): string | undefined {
+  const isUnique = element.id && element.ownerDocument.body.querySelectorAll(`#${element.id}`).length === 1
+
+  if (isUnique) return `#${element.id}`
+}
+
+function getClassSelector(element: Element): string | undefined {
+  const orderedClassList = arrayFrom(element.classList).sort()
+  const siblings = arrayFrom(element.parentElement!.children).filter((child) => child !== element)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Don't use `arrayFrom` so much, as it has performances implications:
* for `orderedClassList`, use `arrayFrom`/`sort` only if you are actually using it (in the `classUniqueAmongSiblings` case)
* for `siblings`, use a for loop over children, you don't need an array for it